api/src/main/java/org/pmiops/workbench/api/CohortsController.java

```diff
@@ -179,19 +196,57 @@ public Cohort apply(org.pmiops.workbench.db.model.Cohort cohort) {
   @Override
   public ResponseEntity<MaterializeCohortResponse> materializeCohort(String workspaceNamespace,
       String workspaceId, MaterializeCohortRequest request) {
-    // TODO(danrodney): get list of participant IDs by:
-    // 1. Retrieve participant cohort statuses matching the status filter.
-    // 2. If the status filter does not contain NOT_REVIEWED, or the cohort review contains
-    // all participant IDs in the cohort, return the IDs directly (subject to pagination.)
-    // 3. Otherwise, query BigQuery for participant IDs using SQL constructed from the cohort
-    // criteria, subject to pagination; remove IDs for participants that had a cohort status not
-    // included in the status filter.
-    return ResponseEntity.ok(new MaterializeCohortResponse());
+    Workspace workspace = workspaceService.getRequired(workspaceNamespace, workspaceId);
+    CdrVersion cdrVersion = workspace.getCdrVersion();
+    if (request.getCdrVersionName() != null) {
+      cdrVersion = cdrVersionDao.findByName(request.getCdrVersionName());
+      if (cdrVersion == null) {
+        throw new BadRequestException(String.format("Couldn't find CDR version with name %s",
+            request.getCdrVersionName()));
+      }
+    }
+    String cohortSpec;
+    if (request.getCohortName() != null) {
+      org.pmiops.workbench.db.model.Cohort cohort =
+          cohortDao.findCohortByNameAndWorkspaceId(request.getCohortName(), workspace.getWorkspaceId());
+      if (cohort == null) {
+        throw new BadRequestException(
+            String.format("Couldn't find cohort with name %s in workspace %s/%s",
+                request.getCohortName(), workspaceNamespace, workspaceId));
+      }
+      cohortSpec = cohort.getCriteria();
+    } else if (request.getCohortSpec() != null) {
+      cohortSpec = request.getCohortSpec();
+    } else if (true){
+      throw new BadRequestException("Must specify either cohortName or cohortSpec");
+    }
+    Integer pageSize = request.getPageSize();
+    if (pageSize == null) {
+      pageSize = DEFAULT_PAGE_SIZE;
+    } else {
+      if (pageSize < 1 || pageSize > MAX_PAGE_SIZE) {
+        throw new BadRequestException(
+            String.format("Invalid page size: %s; must be between 1 and %d", pageSize,
+                MAX_PAGE_SIZE));
+      }
+    }
+
+    SearchRequest searchRequest;
+    try {
+      searchRequest = new Gson().fromJson(cohortSpec, SearchRequest.class);
```

Comment:
Now that this type exists (I think it didn't before?) can materializeCohort just refer to this type directly?